File: 백준/Silver/10845. 큐/큐.swift
import Foundation

let n = Int(readLine()!)!
var queue:[Int] = []

for _ in 0..<n {
    let cmd = readLine()!.split(separator: " ").map{String($0)}
    switch (cmd[0]) {
        case "push": queue.append(Int(cmd[1])!)

        case "pop":
        if (queue.isEmpty) { print(-1) }
        else { print(queue.removeFirst()) }
        
        case "size": print(queue.count)

        case "empty": print(queue.isEmpty ? 1 : 0)
        
        case "front":
        if (queue.isEmpty) { print(-1) }
        else { print(queue[0]) }
        
        case "back":
        if (queue.isEmpty) { print(-1) }
        else { print(queue[queue.count-1]) }

        default: break
    }
}

File: 백준/Silver/10845. 큐/큐.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Queue<Integer> queue = new LinkedList<>();

        int N = sc.nextInt();
        int last = -1;
        for(int i=0; i<N; i++) {
            switch(sc.next()) {
                case "push":
                    last = sc.nextInt();
                    queue.offer(last);
                    break;
                case "pop":
                    System.out.println(queue.isEmpty() ? -1 : queue.poll());
                    break;
                case "size":
                    System.out.println(queue.size());
                    break;
                case "empty":
                    System.out.println(queue.isEmpty() ? 1 : 0);
                    break;
                case "front":
                    System.out.println(queue.isEmpty() ? -1 : queue.peek());
                    break;
                case "back":
                    System.out.println(queue.isEmpty() ? -1 : last);
                    break;
            }
        }
    }
}